lib/PlacesProvider.js

```diff
@@ -134,6 +134,53 @@ Links.prototype = {
   },
 
   /**
+   * Gets the top recent links
+   *
+   * @param {Integer} options
+   *                  limit: Maximum number of results to return. Max 100.
+   *
+   * @returns {Promise} Returns a promise with the array of links as payload.
+   */
+  getTopRecentLinks: Task.async(function*(options) {
+
+    if (!options) {
+      options = {};
+    }
+
+    let {limit} = options;
+    if (!limit || limit.options > HISTORY_RESULTS_LIMIT) {
+      limit = HISTORY_RESULTS_LIMIT;
+    }
+
+    let sqlQuery = "SELECT moz_places.url, moz_favicons.data as favicon, mime_type, title, " +
+                   "      frecency, last_visit_date as lastVisitDate,  \"history\" as type " +
+                   "FROM moz_places " +
+                   "LEFT JOIN moz_favicons " +
+                   "ON favicon_id = moz_favicons.id " +
+                   "WHERE hidden = 0 AND last_visit_date NOTNULL " +
+                   "ORDER BY lastVisitDate DESC, frecency DESC, url " +
+                   "LIMIT :limit";
+
+    let links = yield this.executePlacesQuery(sqlQuery, {
+                  columns: ["url", "title", "lastVisitDate", "frecency"],
+                  params: {limit: limit}
+                });
+
+    // @TODO - this the exact same code as getTopFrecentSites uses
```

Comment:
Do not leave this as a TODO. In fact, the function is extracted in #110. Rebase on top when it is merged?